### PR TITLE
wal: fix CRC corruption on writes following write tears

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

File: pkg/fileutil/fileutil_test.go [hunk 1]
```diff
@@ -118,3 +118,42 @@ func TestExist(t *testing.T) {
...
+	off, serr := f.Seek(0, os.SEEK_CUR)
+	if serr != nil {
+		t.Fatal(serr)
+	}
+	if off != 512 {
+		t.Fatalf("expected offset 512, got %d", off)
+	}
+
+	b = make([]byte, 512)
+	if _, err = f.Read(b); err != nil {
```
Comment (gyuho) [inline pkg/fileutil/fileutil_test.go]: Is this for reading the next 512 bytes from `os.SEEK_CUR`?

Comment (heyitsanthony) [inline pkg/fileutil/fileutil_test.go]: yeah, it's for checking that everything past SEEK_CUR was actually zero'd out

File: wal/wal_test.go [hunk 2]
```diff
@@ -636,3 +636,89 @@ func TestRestartCreateWal(t *testing.T) {
...
+
+	fn := w.tail().Name()
+	w.Close()
+
+	// clobber some entry with 0's to simulate a torn write
+	f, ferr := os.OpenFile(fn, os.O_WRONLY, fileutil.PrivateFileMode)
+	if ferr != nil {
+		t.Fatal(ferr)
+	}
+	defer f.Close()
```
Comment (gyuho) [inline wal/wal_test.go]: We already `f.Close` below?

Comment (heyitsanthony) [inline wal/wal_test.go]: not on the `Fatal` paths

Comment (gyuho) [inline wal/wal_test.go]: oh nvm. you are right.

**Comment by gyuho** [comment]
lgtm. Thanks!

Defer to @xiang90

**Comment by xiang90** [comment]
LGTM. 

Before merging this, @gyuho can you run this patch with failpoint enabled and qps=2500?

**Comment by gyuho** [comment]
@xiang90 Ok let me test run this with failpoints.

**Comment by xiang90** [comment]
@heyitsanthony Let's get this merged! Confirmed this fixed the wal corruption issue.

**Pull request merged**